internal/eval/hmap_test.go

```diff
@@ -116,4 +116,14 @@ func TestHashMapIncrementFloatValue(t *testing.T) {
 	val, err = hmap.incrementFloatValue("field2", 1.0)
 	assert.NotNil(t, err, "Expected error when incrementing a non-float value")
 	assert.Equal(t, errors.IntOrFloatErr, err.Error(), "Expected int or float error")
+
+	inf := math.MaxFloat64
+
+	val, err = hmap.incrementFloatValue("field1", inf+float64(1e308))
```

Comment:
I just replaced this line with the below code and the test fails suggesting that the operation is giving no errors for `field4` . Is this expected behaviour or am I missing something? 
```
val, err = hmap.incrementFloatValue("field4", 1.0)
val, err = hmap.incrementFloatValue("field4", math.MaxFloat64)
```

Reply:
Yes, this is the expected behavior.
https://forums.swift.org/t/what-is-the-float-overflow-behavior/33676/3